src/buffer_advance.rs

```diff
@@ -0,0 +1,85 @@
+use std::{mem::ManuallyDrop, path::PathBuf};
+
+use clap::Parser;
+use vector_buffers::{
+    disk_v2::{Buffer, DiskBufferConfigBuilder},
+    BufferUsageHandle, EventCount as _,
+};
+use vector_common::finalization::Finalizable;
+use vector_core::event::{EventArray, EventStatus};
+
+#[derive(Parser, Debug)]
+#[clap(rename_all = "kebab-case")]
+pub struct Opts {
+    /// Specifies the record ID number to advance over. Without this option, this program just
+    /// outputs the current ledger state. If the first record does not have this ID number, this
+    /// program does nothing.
+    #[clap(long)]
+    record_id: Option<u64>,
+    /// The directory in which the disk buffer resides.
+    data_dir: PathBuf,
+}
+
+#[allow(clippy::print_stdout, clippy::print_stderr)]
+pub(crate) async fn cmd(opts: &Opts) -> exitcode::ExitCode {
+    let config = match DiskBufferConfigBuilder::from_path(&opts.data_dir).build() {
+        Ok(config) => config,
+        Err(error) => {
+            eprintln!("Could not build disk buffer config: {}", error);
+            return exitcode::CONFIG;
+        }
+    };
+
+    let (_writer, mut reader, ledger) = match Buffer::<EventArray>::from_config_inner(
+        config,
+        BufferUsageHandle::noop(),
+        true,
+    )
+    .await
+    {
+        Ok(buffer) => buffer,
+        Err(error) => {
+            eprintln!("Could not open disk buffer: {}", error);
+            return exitcode::IOERR;
+        }
+    };
+
+    let state = ledger.state();
+    println!(
```

Comment:
Just a note, but: we should use the real logging macro (`info!`, `error!`, etc) here instead of `println!`. Not sure if it was already on your list or not, though.

Reply:
Addressed in ed0e8d692